src/main/java/org/brapi/test/BrAPITestServer/service/germ/GermplasmService.java

```diff
@@ -161,9 +151,164 @@ public Page<GermplasmEntity> findGermplasmEntities(@Valid GermplasmSearchRequest
 				.appendList(request.getFamilyCodes(), "familyCode");
 
 		Page<GermplasmEntity> page = germplasmRepository.findAllBySearch(searchQuery, pageReq);
+
+		if(!page.isEmpty()) {
+			log.debug("fetching xrefs");
+			fetchXrefs(page);
+			log.debug("fetching additionalInfo");
+			fetchAdditionalInfo(page);
+			log.debug("fetching attributes");
+			fetchAttributes(page);
+			log.debug("fetching donors");
+			fetchDonors(page);
+			log.debug("fetching origins");
+			fetchOrigin(page);
+			log.debug("fetching institutes");
+			fetchInstitutes(page);
+			log.debug("fetching taxons");
+			fetchTaxons(page);
+			log.debug("fetching storage codes");
+			fetchStorageCodes(page);
+			log.debug("fetching pedigree edges");
+			fetchPedigreeEdges(page);
+		}
+
 		return page;
 	}
 
+	private void fetchXrefs(Page<GermplasmEntity> page) {
+		SearchQueryBuilder<GermplasmEntity> searchQuery = new SearchQueryBuilder<GermplasmEntity>(GermplasmEntity.class);
+		searchQuery.leftJoinFetch("externalReferences", "externalReferences")
+				.leftJoinFetch("pedigree", "pedigree")
+				   .appendList(page.stream().map(BrAPIBaseEntity::getId).collect(Collectors.toList()), "id");
+
+		Page<GermplasmEntity> xrefs = germplasmRepository.findAllBySearch(searchQuery, PageRequest.of(0, page.getSize()));
+
+		Map<String, List<ExternalReferenceEntity>> xrefByEntity = new HashMap<>();
+		xrefs.forEach(entity -> xrefByEntity.put(entity.getId(), entity.getExternalReferences()));
+
+		page.forEach(entity -> entity.setExternalReferences(xrefByEntity.get(entity.getId())));
+	}
+
+	private void fetchAdditionalInfo(Page<GermplasmEntity> page) {
+		SearchQueryBuilder<GermplasmEntity> searchQuery = new SearchQueryBuilder<GermplasmEntity>(GermplasmEntity.class);
+		searchQuery.leftJoinFetch("additionalInfo", "additionalInfo")
+				   .leftJoinFetch("pedigree", "pedigree")
```

Comment:
Why is the LEFT JOIN FETCH with pedigree needed here also?

Reply:
Same as the other area, not entirely sure...when it wasn't there, then Hibernate was executing a select query against the pedigree table for each object in the result set.  Adding this leftJoinFetch resolved that.